cpp/walking-coverage.cpp

```diff
@@ -964,15 +994,30 @@ int main(int argc, char *argv[])
   float nodata;
   unique_ptr<float[]> friction = load_friction_data(frictionRaster.dataset(), 1, &width, &height, &nodata);
 
-  const int maxTimeCost = options._maxTimeCost; // minutes
+  const int maxTimeCost = options._maxTimeCost[0]; // minutes
+
   unique_ptr<float[]> cost =
     run_dijkstra_on_friction_layer(friction.get(),
                                    width, height, nodata,
                                    frictionRaster.pixel_width_meters(),
                                    frictionRaster.pixel_height_meters(),
                                    pixelOrigin.first, pixelOrigin.second,
                                    maxTimeCost,
-                                   options._minFriction);
+                                   options._minFriction[0]);
+
+  for (size_t i = 1; i < options._maxTimeCost.size(); ++i) {
+    unique_ptr<float[]> new_cost =
+      run_dijkstra_on_friction_layer(friction.get(),
+                                     width, height, nodata,
+                                     frictionRaster.pixel_width_meters(),
+                                     frictionRaster.pixel_height_meters(),
+                                     pixelOrigin.first, pixelOrigin.second,
+                                     options._maxTimeCost[i],
+                                     options._minFriction[i]);
+
+    // Merging layers scale costs to make it is easier to calculate the isochrone
```

Comment:
This comment is a bit misleading. I'd say you need to adjust the new cost layer to take it to the same scale as the original in order to compute the isochrone.

Reply:
I'll write a better explanation.